apps/test/unit/blockly/customFunctionsTest.js

```diff
@@ -0,0 +1,72 @@
+/* global Blockly */
+import sinon from 'sinon';
+import GoogleBlockly from 'blockly/core';
+import initializeGoogleBlocklyWrapper from '@cdo/apps/blockly/googleBlocklyWrapper';
+import {expect} from '../../util/reconfiguredChai';
+import '@cdo/apps/flappy/flappy'; // Importing the app forces the test to load Blockly
+import {
+  newDefinitionBlock,
+  allCallBlocks
+} from '../../../src/blockly/addons/functionBlocks.js';
+
+describe('Custom Functions', () => {
+  const cdoBlockly = Blockly;
+  // Reset context menu registry.
+  const registry = JSON.parse(
+    JSON.stringify(GoogleBlockly.ContextMenuRegistry.registry.registry_)
+  );
+  beforeEach(() => {
+    GoogleBlockly.JavaScript = sinon.spy();
+    Blockly = initializeGoogleBlocklyWrapper(GoogleBlockly); // eslint-disable-line no-global-assign
+  });
+  afterEach(() => {
+    // Reset Blockly for other tests.
+    Blockly = cdoBlockly; // eslint-disable-line no-global-assign
+    // Reset context menu for other tests.
+    GoogleBlockly.ContextMenuRegistry.registry.registry_ = JSON.parse(
+      JSON.stringify(registry)
+    );
+  });
+
+  it('Can create a blank function definition block', () => {
+    const defBlock = newDefinitionBlock();
+    const defBlockHTML = defBlock.map(element => element.outerHTML).join();
```

Comment:
Noted re: XML. I'll update.
And yes, I was trying to use a standard format thinking that it might help with readability. Happy to simplify it here though!